File: JavaScript/08. Functions/Functions-Homework/Functions-Homework/02. ReverseNumber/ReverseNumber.js
/*
 * Problem 2. Reverse number

Write a function that reverses the digits of given decimal number.
 */

var arr = [256, 123.45],
    i,
    str = '',
    len;

for (i = 0, len = arr.length; i < len; i += 1) {
    console.log(ReverseNumber(arr[i]));
}

function ReverseNumber(number) {
    str = number.toString();
    var strResult = "";
    for (var j = str.length - 1; j >= 0; j -= 1) {
        strResult += str[j];
    }
    return parseFloat(strResult);
}

File: C# II/03. Methods/Methods-Homework/07. ReverseNumber/ReverseNumber.cs
using System;
using System.Text;

class ReverseNumber
{
    /*
      7. Reverse number

        Write a method that reverses the digits of given decimal number.
        example: 
        input 256 output 652
    */

    static decimal ReverseNumberMethod(decimal number)
    {
        StringBuilder numberToString = new StringBuilder(number.ToString());
        Console.WriteLine(numberToString.ToString());
        char change = ' ';
        for(int i = 0; i < numberToString.Length/2; i++)
        {
            change = numberToString[i];
            numberToString[i] = numberToString[numberToString.Length - 1 - i];
            numberToString[numberToString.Length - 1 - i] = change;
        }
        return Convert.ToDecimal(numberToString.ToString());
    }

    static void Main()
    {
        Console.Write("Enter a number: ");
        decimal inputNumber = decimal.Parse(Console.ReadLine());
        Console.WriteLine("The reversed number  is: " + ReverseNumberMethod(inputNumber));
    }
}

